packages/aws-rfdk/lib/core/lib/resource.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  CfnResource,
+  Construct,
+  IResource,
+  RemovalPolicy,
+  ResourceEnvironment,
+  Stack,
+} from '@aws-cdk/core';
+
+/**
+ * Represents a resource in RFDK.
+ */
+export abstract class Resource extends Construct implements IResource {
+  /**
+   * @inheritdoc
+   */
+  public abstract readonly stack: Stack;
+
+  /**
+   * @inheritdoc
+   */
+  public abstract readonly env: ResourceEnvironment;
+
+  /**
+   * This is the same implementation as CDK's from here:
+   * https://github.com/aws/aws-cdk/blob/cd51a5dae1780e34aecd90d85783fb6d3c239903/packages/@aws-cdk/core/lib/resource.ts
+   *
+   * We don't want to extend our classes from their implementation directly due
+   * to how the CDK Resource is meant to map to L1 constructs.
+   *
+   * Apply the given removal policy to this resource
+   *
+   * The Removal Policy controls what happens to this resource when it stops
+   * being managed by CloudFormation, either because you've removed it from the
+   * CDK application or because you've made a change that requires the resource
+   * to be replaced.
+   *
+   * The resource can be deleted (`RemovalPolicy.DESTROY`), or left in your AWS
+   * account for data recovery and cleanup later (`RemovalPolicy.RETAIN`).
+   */
+  public applyRemovalPolicy(policy: RemovalPolicy) {
```

Comment:
If all Resources in CDK are L1 constructs, then is this implementation correct for RFDK (where our constructs are L3, generally)? Specifically, our constructs can (and do) have more than one child; I wonder if it's actually sufficient to only look at the defaultChild...

Reply:
After speaking with Josh, he raised the question of why some of our constructs even needed to implement `IResource`. It doesn't seem like they do. I've removed the `Resource` construct I wrote and removed the `implements IResource` from all our classes that were using it. I was able to build and synthesize all our examples, so it seems like we don't need to.

Let me know what you think, the only concern I have with the update I made is that I removed the `stack` and `env` properties along with the interface, since they were requirements of the `IResource` interface. They can be retrieved for any construct without needing to store them as construct properties, and it seems really unlikely to me that anybody would have been using them, but it's a breaking change to remove them.

Reply:
I've run the integration tests against the updated code and also packaged it and dropped it into a StudioBuilder app and made sure it could synth without any issues.

Reply:
Sounds good to me. I don't recall why we made them `IResource` instead of `IConstruct` originally.

Reply:
Like you said, it is very unlikely that anyone used these properties, but do you think we should mark this as an official breaking change?

If so, then we could add upgrade documentation and before merging we should squash and edit the commit message to include a conventional commits breaking change. What do you think?